### PR TITLE
The Image post API is done

### PR DESCRIPTION
**Pull request by bc-davin**
Will need testing and some feedback!

### CODE REVIEW

File: posts/views.py
Comment (sruduke) [inline posts/views.py:1]:
```
    Line 263
    # Check if the content is base64-encoded
    if not pic_post.content.startswith('data:'):
        return HttpResponse(status=404)
    # Extract the base64-encoded image data
    image_data = pic_post.content.split(';base64,')[-1]
```
I don't believe this is a valid check for base 64 encoding, as base 64 encoded data, `pic_post.content`, appears like: 
`iVBORw0KGgoAAAANSUhEUgAAAQAAAAEACAIAAADTED8xAAAAAXNSR0IArs4c6QAAIABJREFUeJycvdmWJElyJXaviJp7RORSC6obwI`
Thus, it also doesn't have the ";base64" that you're splitting, did you mean to reference the `pic_post.contentType`? 

Regardless, if the image post exists, all you need to do is base 64 decode the `pic_post.content` -- you don't have to do any extra checks or formatting before.

**Review by sruduke — COMMENTED** [review]
Good work overall, just some small adjustments :)

**Comment by bc-davin** [comment]
Thanks for the review Sophia will adjust to the feedback!

**Comment by sruduke** [comment]
Of course Davin, thanks for the good work!

**Pull request merged**